src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/SPARQL2GraphQLTranslatorImpl.java

```diff
@@ -0,0 +1,454 @@
+                    if(!sgpNodes.containsKey(subject)){
+                        sgpNodes.put(subject,new SGPNode());
+                    }
+                    if(!sgpNodes.containsKey(object)){
+                        sgpNodes.put(object,new SGPNode());
+                    }
+                    final SGPNode subjectSgpNode = sgpNodes.get(subject);
+                    final SGPNode objectSgpNode = sgpNodes.get(object);
+                    subjectSgpNode.addAdjacentNode(tp.getId(),objectSgpNode);
+                    connectors.put(tp.getId(), object);
+                }
+            }
+        }
+        // Remove all potential cyclic connector bindings
+        final Set<Integer> connectorsToBeRemoved = GraphCycleDetector.determineCyclicConnectors(sgpNodes);
+        for(final int i : connectorsToBeRemoved){
+            connectors.remove(i);
+        }
+
+        // Creating necessary variables for the GraphQL query
+        final TreeSet<String> fieldPaths = new TreeSet<>();
+        final JsonObject argumentValues = new JsonObject();
+        final Map<String, String> argumentDefinitions = new HashMap<>();
+
+        // Get all SGP without a connector, if they have undeterminable GraphQL type then materializeAll
+        final Map<Node,String> withoutConnector = new HashMap<>();
+        boolean isDeterminable = true;
+        for(final Node n : subgraphPatterns.keySet()){
+            final String sgpType = determineSgpType(subgraphPatterns.get(n),config);
+            if(!connectors.values().contains(n) && sgpType == null){
+                materializeAll(fieldPaths,subgraphPatterns,connectors,config,endpoint);
+                isDeterminable = false;
+                break;
+            }
+            if(!connectors.values().contains(n)){
+                withoutConnector.put(n,sgpType);
+            }
+        }
+
+        // If the entire view doesn't need to be materialized
+        if(isDeterminable){
+            int entrypointCounter = 0;
+            int variableCounter = 0;
+            for(final Node n : withoutConnector.keySet()){
+                final Map<String,LiteralLabel> sgpArguments = getSgpArguments(subgraphPatterns.get(n),config);
+                final String sgpType = withoutConnector.get(n);
+                final StringBuilder currentPath = new StringBuilder();
+                GraphQLEntrypoint e;
+
+                if(hasNecessaryArguments(sgpArguments.keySet(), 
+                        endpoint.getEntrypoint(sgpType,GraphQLEntrypointType.SINGLE).getArgumentDefinitions().keySet(), 
+                        true)){
+                    // Get single object entrypoint
+                    e = endpoint.getEntrypoint(sgpType,GraphQLEntrypointType.SINGLE);
+                    final StringBuilder entrypointAlias = new StringBuilder("ep_single")
+                    .append(entrypointCounter).append(":").append(e.getFieldName());
```

Comment:
maybe consider adding a method to the Entrypoint class that, given a fieldname and counter returns this alias